server/src/main/java/org/opensearch/rest/action/admin/cluster/RestPutDecommissionAction.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.admin.cluster;
+
+import org.opensearch.action.admin.cluster.decommission.awareness.put.PutDecommissionRequest;
+import org.opensearch.client.Requests;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.cluster.decommission.DecommissionAttribute;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.rest.BaseRestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.action.RestToXContentListener;
+
+import java.io.IOException;
+import java.util.List;
+
+import static java.util.Collections.singletonList;
+import static org.opensearch.rest.RestRequest.Method.PUT;
+
+public class RestPutDecommissionAction extends BaseRestHandler {
+
+    private static final TimeValue DEFAULT_TIMEOUT = TimeValue.timeValueSeconds(300L);
+
+    @Override
+    public List<Route> routes() {
+        return singletonList(new Route(PUT, "/_cluster/decommission/awareness/{awareness_attribute_name}/{awareness_attribute_value}"));
```

Comment:
One thing is not clear from URL is: what is getting decommissioned? Is it a cluster, awareness attribute or nodes with the awareness attribute?

Reply:
As per the issue #3639, implemented the URL. 

> what is getting decommissioned? Is it a cluster, awareness attribute or nodes with the awareness attribute?

Nodes with the given awareness attribute will be removed from the cluster.